### PR TITLE
Switch to pyproject.toml

### PR DESCRIPTION
**Pull request by dstansby**
This updates the packaging to use `pyproject.toml`, following the template at https://github.com/SainsburyWellcomeCentre/python-cookiecutter. This is the first time I've (manually) applied that template to a package, so would appreciate a careful check that I've done everything right.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,10 +1,104 @@
...
+"Bug Tracker" = "https://github.com/brainglobe/bg-atlasapi/issues"
+"Documentation" = "https://docs.brainglobe.info/bg-atlasapi"
+
+[project.optional-dependencies]
+dev = [
+    "black",
+    "bump2version",
+    "check-manifest",
+    "coverage>=5.0.3",
+    "flake8",
```
Comment (adamltyson) [inline pyproject.toml:48]: This can be removed.

File: pyproject.toml [hunk 2]
```diff
@@ -1,10 +1,104 @@
...
+[project.urls]
+"Homepage" = "https://brainglobe.info/atlas-api"
+"Source Code" = "https://github.com/brainglobe/bg-atlasapi"
+"Bug Tracker" = "https://github.com/brainglobe/bg-atlasapi/issues"
+"Documentation" = "https://docs.brainglobe.info/bg-atlasapi"
+
+[project.optional-dependencies]
+dev = [
+    "black",
+    "bump2version",
```
Comment (adamltyson) [inline pyproject.toml:45]: This can be removed.

File: pyproject.toml [hunk 3]
```diff
@@ -1,10 +1,104 @@
...
+
+[tool.setuptools]
+include-package-data = true
+
+[tool.setuptools.packages.find]
+include = ["bg_atlasapi*"]
+exclude = ["tests*"]
+
+[tool.pytest.ini_options]
+addopts = "--cov={{cookiecutter.module_name}}"
```
Comment (adamltyson) [inline pyproject.toml:74]: This should specify the module name.

File: pyproject.toml [hunk 4]
```diff
@@ -1,10 +1,104 @@
...
+readme = "README.md"
+license = {file = "LICENSE"}
+authors = [
+    {name = "Luigi Petrucco, Federico Claudi, Adam Tyson", email = "code@adamltyson.com"},
+]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Operating System :: MacOS :: MacOS X",
```
Comment (adamltyson) [inline pyproject.toml:13]: This could be simplified to `"Operating System :: OS Independent"`

File: pyproject.toml [hunk 5]
```diff
@@ -1,10 +1,104 @@
...
+
+[project.urls]
+"Homepage" = "https://brainglobe.info/atlas-api"
+"Source Code" = "https://github.com/brainglobe/bg-atlasapi"
+"Bug Tracker" = "https://github.com/brainglobe/bg-atlasapi/issues"
+"Documentation" = "https://docs.brainglobe.info/bg-atlasapi"
+
+[project.optional-dependencies]
+dev = [
+    "black",
```
Comment (adamltyson) [inline pyproject.toml:44]: We can probably remove this if we're using pre-commit.

File: pyproject.toml [hunk 6]
```diff
@@ -1,10 +1,104 @@
...
+
+[tool.setuptools.packages.find]
+include = ["bg_atlasapi*"]
+exclude = ["tests*"]
+
+[tool.pytest.ini_options]
+addopts = "--cov={{cookiecutter.module_name}}"
+
 [tool.black]
 target-version = ['py38', 'py39', 'py310']
```
Comment (adamltyson) [inline pyproject.toml:77]: Should add py311 here to match above.

File: pyproject.toml [hunk 7]
```diff
@@ -1,10 +1,104 @@
...
 skip-string-normalization = false
 line-length = 79
 
+[tool.setuptools_scm]
+
+[tool.check-manifest]
+ignore = [
+  "*.yaml",
+  ".bumpversion.cfg",
```
Comment (adamltyson) [inline pyproject.toml:86]: Can be removed.

File: pyproject.toml [hunk 8]
```diff
@@ -1,10 +1,104 @@
...
+  "tox.ini",
+  "tests/*",
+  "tests/test_unit/*",
+  "tests/test_integration/*",
+  ".flake8"
```
Comment (adamltyson) [inline pyproject.toml:91]: Can be removed.

**Review by adamltyson — CHANGES_REQUESTED** [review]
Looks good. Added a few small comments.

**Comment by dstansby** [comment]
Should all be fixed - I took the liberty as marking everything as resolved above as they were pretty clear cut changes.

**Review by adamltyson — APPROVED** [review]

**Pull request merged**